### PR TITLE
Update the automated release process description

### PR DESCRIPTION
**Pull request by Jaanki**
...mentioning to merge the PRs open on dependent projects before
 moving to the next step and few mispell corrections.

Signed-Off-By: Janki Chhatbar <jchhatba@redhat.com>

### CODE REVIEW

File: src/content/development/release-process/_index.en.md [hunk 1]
```diff
@@ -99,7 +100,10 @@ To do so, run the `make` command with `dryrun=true`.
 {{% /notice %}}
 
 The command runs, gathers the data for the release, updates the release YAML and pushes it for review. Once the review process is done,
-and the automated release steps on the CI have finished successfully, simply **run the same command again to advance to the next stage**.
+and the automated release steps on the CI have finished successfully, the PR is merged. Pull requests will be created for all dependent
+projects to update them to the new version in preparation for the subsequent  steps. The automation will leave a comment with a list of
+them. Make sure all those PRs are merged and their release jobs pass. Then proceed with the next step, by simply **running the same
```
Comment (tpantelis) [inline src/content/development/release-process/_index.en.md]:
```suggestion
and the automated release steps on the CI have finished successfully, merge the PR. Pull requests will then be created for all dependent
projects to update them to the new version pertaining to the current step, if necessary, in preparation for the next step. The automation will also leave a comment with a list of
the dependent project PRs in the release PR . Make sure all those PRs are merged and their release jobs pass then proceed to the next step, by simply **running the same
```

**Review by tpantelis — CHANGES_REQUESTED** [review]

File: src/content/development/release-process/_index.en.md [hunk 2]
```diff
@@ -36,19 +36,20 @@ Version numbers are required to be formatted following the schema norms where th
...
+The release process is mostly automated and uses a YAML file created in the
+[releases repository](https://github.com/submariner-io/releases) that describes the release.
 This file is updated for each step in the release process.
 
 Once the changes for a step are reviewed and merged, a CI job will run to create the release(s) for the step and create the
 required pull requests in preparation for the next step to be reviewed and merged. Once all these pull requests have been
 merged, you can continue onto the next step.
 
-For most projects, after a release is created, another job will be initiated to build release artifacts and publish to Quay.
+For most projects, after a release is created, another job will be initiated to build release artifacts and publish it to Quay.
```
Comment (dfarrell07) [inline src/content/development/release-process/_index.en.md]:
```suggestion
For most projects, after a release is created, another job will be initiated to build release artifacts and publish them to Quay.
```

I also think it's okay as-is, "publish to Quay".

File: src/content/development/release-process/_index.en.md [hunk 3]
```diff
@@ -36,19 +36,20 @@ Version numbers are required to be formatted following the schema norms where th
...
 This will take several minutes. You can monitor the progress from the project's main page. In the branches/tags pull-down
 above the file list heading, select the tag for the new version. A small yellow circle icon should be present to the right
-of the file list heading which indicates a job is in progress. You can click it to see details. There may be several checks
+of the file list heading which indicates a job is in progress. You can click on it to see details. There may be several checks
```
Comment (dfarrell07) [inline src/content/development/release-process/_index.en.md]: The guidance I'm seeing is that it should remain as-is.

https://helpcenter.veeam.com/docs/styleguide/tw/click.html

File: src/content/development/release-process/_index.en.md [hunk 4]
```diff
@@ -99,7 +100,10 @@ To do so, run the `make` command with `dryrun=true`.
 {{% /notice %}}
 
 The command runs, gathers the data for the release, updates the release YAML and pushes it for review. Once the review process is done,
-and the automated release steps on the CI have finished successfully, simply **run the same command again to advance to the next stage**.
+and the automated release steps on the CI have finished successfully, merge the PR. Pull requests will then be created for all dependent
+projects to update them to the new version pertaining to the current step, if necessary, in preparation for the next step. The automation
```
Comment (dfarrell07) [inline src/content/development/release-process/_index.en.md]:
```suggestion
projects to update them to the new version. The automation
```

File: src/content/development/release-process/_index.en.md [hunk 5]
```diff
@@ -99,7 +100,10 @@ To do so, run the `make` command with `dryrun=true`.
 {{% /notice %}}
 
 The command runs, gathers the data for the release, updates the release YAML and pushes it for review. Once the review process is done,
-and the automated release steps on the CI have finished successfully, simply **run the same command again to advance to the next stage**.
+and the automated release steps on the CI have finished successfully, merge the PR. Pull requests will then be created for all dependent
```
Comment (dfarrell07) [inline src/content/development/release-process/_index.en.md]:
```suggestion
, merge the PR. Pull requests will then be created for all dependent
```

The original meant the automation that runs after the PR is merged, your change makes it about the tests on the PR. I think the result is a bit redundant, we don't need to explain that tests have to pass as a part of the review process and you talk about the post-merge automation later.

File: src/content/development/release-process/_index.en.md [hunk 6]
```diff
@@ -99,7 +100,10 @@ To do so, run the `make` command with `dryrun=true`.
...
+projects to update them to the new version pertaining to the current step, if necessary, in preparation for the next step. The automation
+will also leave a comment with a list of the dependent project PRs in the release PR . Make sure all those PRs are merged and their
```
Comment (dfarrell07) [inline src/content/development/release-process/_index.en.md]:
```suggestion
will leave a comment with a list of the version-bump PRs for dependent project the release PR that was just merged. Make sure all those PRs are merged and their
```

File: src/content/development/release-process/_index.en.md [hunk 7]
```diff
@@ -99,7 +100,10 @@ To do so, run the `make` command with `dryrun=true`.
...
+release jobs pass then proceed to the next step, by simply **running the same command again to advance to the next stage**.
```
Comment (dfarrell07) [inline src/content/development/release-process/_index.en.md]:
```suggestion
release jobs pass (see the Actions tab of the repository on GitHub) then **proceed to the next release phase by running the same command again**.
```

**Review by dfarrell07 — CHANGES_REQUESTED** [review]

File: src/content/development/release-process/_index.en.md (hunk 1 above)
Comment (dfarrell07) [reply src/content/development/release-process/_index.en.md]: > pertaining to the current step, if necessary, in preparation for the next step

Sorry to have conflicting advice for this bit @Jaanki, but I really think it's unnecessary, that it adds cognitive load without adding anything useful.

**Comment by nyechiel** [comment]
@Jaanki ping ^^

File: src/content/development/release-process/_index.en.md (hunk 2 above)
Comment (Jaanki) [reply src/content/development/release-process/_index.en.md]: Reverting to what it was originally.

**Review by dfarrell07 — APPROVED** [review]
LGTM, just has a few MD linting errors.

**Review by sridhargaddam — APPROVED** [review]

**Review by tpantelis — APPROVED** [review]

**Pull request merged**